src/test/java/com/gfl/client/model/ProxyConfigHolderTest.java

```diff
@@ -0,0 +1,195 @@
+        credentials1.setUsername("username1");
+        credentials1.setPassword("password1");
+        configHolder1.setProxyCredentials(credentials1);
+        configHolder1.setProxyNetworkConfig(config1);
+
+        ProxyConfigHolder configHolder2 = new ProxyConfigHolder();
+        ProxyNetworkConfig config2 = new ProxyNetworkConfig();
+        config2.setHostname("hostname2");
+        config2.setPort(8082);
+        ProxyCredentials credentials2 = new ProxyCredentials();
+        credentials2.setUsername("username2");
+        credentials2.setPassword("password2");
+        configHolder2.setProxyCredentials(credentials2);
+        configHolder2.setProxyNetworkConfig(config2);
+
+        return Stream.of(configHolder1, configHolder2);
+    }
+
+    @Test
+    @DisplayName("Test hashCode method for proxyConfigHolder object")
+    void testHashCode() {
+        int expected = proxyConfigHolder.hashCode();
+        assertEquals(expected, proxyConfigHolder.hashCode());
+    }
+
+    @Test
+    @DisplayName("Test toString method for proxyConfigHolder object")
+    void testToString() {
+        String expected = "ProxyConfigHolder(proxyNetworkConfig=" +
+                "ProxyNetworkConfig(hostname=hostname, port=8080)," +
+                " proxyCredentials=ProxyCredentials(username=username, password=password))";
+        assertEquals(expected, proxyConfigHolder.toString());
+    }
+
+    @Test
+    @DisplayName("Test equality and hashCode for   ProxyConfigHolder objects with the same values")
+    void testEqualsAndHashCode() {
+        ProxyConfigHolder proxyConfigHolder1 = new ProxyConfigHolder();
+        ProxyNetworkConfig config = new ProxyNetworkConfig();
+        config.setHostname("hostname");
+        config.setPort(8080);
+        ProxyCredentials credentials = new ProxyCredentials();
+        credentials.setUsername("username");
+        credentials.setPassword("password");
+
+        proxyConfigHolder1.setProxyCredentials(credentials);
+        proxyConfigHolder1.setProxyNetworkConfig(config);
+
+        assertEquals(proxyConfigHolder, proxyConfigHolder1);
+        assertEquals(proxyConfigHolder.hashCode(), proxyConfigHolder1.hashCode());
+
+    }
+
+    @Test
+    @DisplayName("Test validation fails for null proxyNetworkConfig")
+    void testValidationFailsForNullProxyNetworkConfig() {
+        ProxyConfigHolder configHolder = new ProxyConfigHolder(null, proxyCredentials);
+
+        Set<ConstraintViolation<ProxyConfigHolder>> violations = validator.validate(configHolder);
+
+        // Assert violations for null proxyNetworkConfig
+        assertEquals(1, violations.size());
+        ConstraintViolation<ProxyConfigHolder> violation = violations.iterator().next();
+        assertEquals("proxy network configuration can't be null", violation.getMessage());
+        assertEquals("proxyNetworkConfig", violation.getPropertyPath().toString());
+    }
+
+    @Test
+    @DisplayName("Test validation fails for invalid proxyCredentials")
```

Comment:
This test contains only 3 test cases, 
Better use some MethodSource to cover valid and invalid cases based on traceability matrix

```
private static Stream<Object[]> invalidProxyConfigHolder() { }, 
private static Stream<Object[]> validProxyConfigHolder() { }
```